Add Useful Resources and Example Build Instructions to CONTRIBUTING.md

HudsonHumphries opened the pull request:
These are resources I found helpful as I was getting to learn about Open Telemetry and specifically the C++ repository. Adding these resources will help newcomers as they will have a central place to go to for resources. This Open Telemetry Onboarding Resources Guide was written by @hudsonhumphries, @ankit-bhargava and @Brandon-Kimberly

reyang commented:
> I wonder about merging this with the [CONTRIBUTING.md](https://github.com/open-telemetry/opentelemetry-cpp/blob/master/CONTRIBUTING.md).

+1

HudsonHumphries commented:
Thanks for the feedback! I am working with Ankit who made [PR 94](https://github.com/open-telemetry/opentelemetry-cpp/pull/94) to move our documentation into CONTRIBUTING.md.

g-easy commented on docs/open-telemetry-onboarding-resource-guide.md:
```diff
@@ -0,0 +1,45 @@
+* Read through the [Open Telemetry Collector](https://github.com/open-telemetry/opentelemetry-collector) GitHub repository
+    * This repository has a lot of good information surrounding the Open Telemetry ecosystem. At the top of the **[readme](https://github.com/open-telemetry/opentelemetry-collector/blob/master/README.md)**, there are multiple links that give newcomers a good idea of what the project is about and how to get involved in it.
+* Read through the Open Telemetry Python documentation for its [API](https://opentelemetry-python.readthedocs.io/en/stable/api/api.html) and [SDK](https://opentelemetry-python.readthedocs.io/en/stable/sdk/sdk.html)
+    * The [API](https://github.com/open-telemetry/opentelemetry-java/blob/master/QUICKSTART.md) and [SDK](https://github.com/open-telemetry/opentelemetry-java/blob/master/QUICKSTART.md) documentation provides a lot of information on what the classes and their functions are used for. Since there is currently minimal documentation for C++, use the Python repository’s extensive documentation to lear ...
```
Link is to Java but no mention of Java in the text?

HudsonHumphries replied:
I believe those should be linking to the API and SDK documentation for Python, thanks for catching that. Will fix it

HudsonHumphries commented:
As Ankit mentioned in his PR, we moved build instructions for examples into this PR. I also moved the resources into CONTRIBUTING.md and fixed bad links.

reyang commented on CONTRIBUTING.md:
```diff
@@ -72,3 +72,60 @@ A PR is considered to be **ready to merge** when:
+* Take a look at this [Java SDK example.](https://github.com/open-telemetry/opentelemetry-java/tree/master/examples/sdk-usage) This shows a good use case of the SDK using stdout exporter.
+* Take a look at the [Java Jaeger example.](https://github.com/open-telemetry/opentelemetry-java/tree/master/examples/jaeger) This provides a brief introduction to the Jaeger exporter, its interface, and how to interact with the service.
+
+Please contribute! You’re welcome to add more information if you come across any helpful resources
```
nit: missing an end period.

reyang commented on CONTRIBUTING.md:
```diff
@@ -72,3 +72,60 @@ A PR is considered to be **ready to merge** when:
 
 Any Approver / Maintainer can merge the PR once it is **ready to merge**.
+
+## Build and Run Code Examples
```
I would suggest putting this section under `Development` (above `How to Send Pull Requests`).

pyohannes commented on CONTRIBUTING.md:
```diff
@@ -72,3 +99,33 @@ A PR is considered to be **ready to merge** when:
+
+### Code Examples
+
+* Follow the [simple trace example](https://github.com/open-telemetry/opentelemetry-cpp/pull/92) for an introduction to basic Open Telemetry functionality in C++.  Currently the example can be found in [PR #94](https://github.com/open-telemetry/opentelemetry-cpp/pull/94)
```
When this is merged, can you submit an issue that reminds us of updating this document before we do a first release? Once we have our own examples and exporters in place, many of the examples here should be switched out.

HudsonHumphries replied:
Yes, I'll get that issue up once this PR this merged

reyang commented:
@pyohannes mentioned that he is okay to merge this PR during the community meeting.
Please rebase, thanks.

HudsonHumphries commented:
Rebased @reyang
